cli/tools/lint/minified_file.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+#[derive(Debug)]
+pub struct FileMetrics {
+  long_lines_count: usize,
+  total_lines: usize,
+  whitespace_ratio: f64,
+  has_license_comment: bool,
+}
+
+impl FileMetrics {
+  #[inline]
+  pub fn is_likely_minified(&self) -> bool {
+    let long_lines_ratio =
+      self.long_lines_count as f64 / self.total_lines as f64;
+
+    (long_lines_ratio >= 0.2 || self.whitespace_ratio < 0.05)
+      && !(self.has_license_comment && self.total_lines < 3)
+  }
+}
+
+/// Analyze the content and tell if the file is most likely a minified file or not.
+pub fn is_likely_minified(content: &str) -> bool {
+  const LONG_LINE_LEN: usize = 250;
+  let mut total_lines = 0;
+  let mut long_lines_count = 0;
+  let mut whitespace_count = 0;
+  let mut total_chars = 0;
+  let mut has_license = false;
+  let mut in_multiline_comment = false;
+
+  // If total len of a file is shorter than the "long line" length, don't bother analyzing
+  // and consider non-minified.
+  if content.len() < LONG_LINE_LEN {
+    return false;
+  }
+
+  // Preallocate a line buffer to avoid per-line allocations
+  let mut line_buffer = String::with_capacity(1024);
+
+  // Process the content character by character to avoid line allocations
+  let mut chars = content.chars().peekable();
+  while let Some(c) = chars.next() {
+    total_chars += 1;
+
+    if c.is_whitespace() {
+      whitespace_count += 1;
+    }
+
+    line_buffer.push(c);
```

Comment:
Maybe we should maintain a string reference instead of allocating and needing to clear a vector?